### PR TITLE
Performance: Progressively render descriptions

### PR DESCRIPTION
**Pull request by andy-hook**
Fixes https://github.com/aragon/network-dashboard/issues/72

Improves our perceived performance by unblocking UI rendering and progressively loading each call to `describeScript`, we also now cache those descriptions in a mapping based on `vote.id` so that we don't have to do the same work multiple times.

ezgif-7-354765a93a5a

### CODE REVIEW

File: src/components/Proposals/ProposalCard.js [hunk 1]
```diff
@@ -69,12 +79,14 @@ function ProposalCard({ vote, onProposalClick }) {
...
         `}
       >
-        <AppBadge
-          label={target.name ? target.name : target.address}
-          appAddress={target.address}
-          iconSrc={target.icon ? target.icon : ''}
-          badgeOnly
-        />
+        {emptyScript ? (
+          <DefaultAppBadge />
```
Comment (andy-hook) [inline src/components/Proposals/ProposalCard.js:83]: This was based on the previous version and is fallback disputable app badge. @delfipolito Are we sure this should be the default app when there is no description available to process?

File: src/components/Proposals/ProposalCard.js [hunk 2]
```diff
@@ -93,11 +105,17 @@ function ProposalCard({ vote, onProposalClick }) {
           overflow: hidden;
         `}
       >
-        <strong css="font-weight: bold">#{voteId}: </strong>
-        {Array.isArray(description) ? (
-          <Description path={description} />
+        {emptyScript ? (
```
Comment (andy-hook) [inline src/components/Proposals/ProposalCard.js:108]: I don't love this boolean check for an empty script, but it was the cleanest way I could think to present the branching logic depending on what data is available.

File: src/components/Proposals/ProposalCard.js [hunk 3]
```diff
@@ -41,7 +45,13 @@ function getAttributes(status, theme) {
...
 function ProposalCard({ vote, onProposalClick }) {
   const theme = useTheme()
-  const { context, voteId, description, target } = vote
+  const { context, voteId, script } = vote
+  const {
+    description,
+    emptyScript,
+    targetApp,
+    loading: descriptionLoading,
+  } = useDescribeVote(script, vote.id)
```
Comment (andy-hook) [inline src/components/Proposals/ProposalCard.js:54]: This hook also returns the target data for the app badge, this feels a bit cleaner to me and we can always pull it out in the future if we need to use it in other contexts.

File: src/components/Proposals/Description.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import React from 'react'
 import { GU, IdentityBadge, useTheme } from '@aragon/ui'
+import { propTypes } from '@aragon/ui/dist/index-46d0e707'
```
Comment (delfipolito) [inline src/components/Proposals/Description.js:3]: Should we have  our own proptypes here or export them properly on aragonUI?

File: src/components/Proposals/ProposalCard.js (hunk 1 above)
Comment (delfipolito) [reply src/components/Proposals/ProposalCard.js:83]: Yes :)

File: src/components/Proposals/Description.js (hunk 4 above)
Comment (andy-hook) [reply src/components/Proposals/Description.js:3]: Ohhh nice spot! my editor auto imported this from the wrong place 🤦  thanks!

File: src/components/Proposals/ProposalCard.js (hunk 1 above)
Comment (andy-hook) [reply src/components/Proposals/ProposalCard.js:83]: Gotcha!

File: src/components/Proposals/Description.js (hunk 4 above)
Comment (andy-hook) [reply src/components/Proposals/Description.js]: Fixed in https://github.com/aragon/network-dashboard/pull/74/commits/f6b10c9365a20c11470edb5595df9e8d67f52e4f

**Review by delfipolito — APPROVED** [review]

**Pull request merged**